mne/preprocessing/faster.py

```diff
@@ -0,0 +1,414 @@
+import numpy as np
+from collections import defaultdict
+from scipy.signal import welch, lfilter
+from scipy.stats import kurtosis
+from . import find_outliers
+from ..utils import logger
+from ..io.pick import pick_types, channel_type
+
+
+def _by_ch_type(info, picks):
+    """Returns lists of channel indices, grouped by channel type."""
+    ch_types = defaultdict(list)
+
+    for ch in picks:
+        ch_types[channel_type(info, ch)].append(ch)
+
+    return ch_types.items()
```

Comment:
.... we have such a function already somewhere in picks, even two of them, one public, one private.